0000-nonlexical-lifetimes.md

```diff
@@ -1575,8 +1579,8 @@ Another example, this time using a `match`:
 ```rust
 fn main() {
     let mut x = Some(3);
-    match &mut x {
-        Some(i) => {
+    match *&mut x {
+        Some(ref mut i) => {
```

Comment:
But your version [doesn't compile](https://play.rust-lang.org/?gist=be9c3547b62d4d9b5e2aae1eeb73b619&version=stable). That's the only reason I fixed it. It's also possible to solve like this:

```rust
match &mut x {
    &mut Some(ref mut i) => { ... }
    &mut None => { ... }
}       
```
But I think that this is discouraged.

Or am I misunderstanding your example and it is supposed to fail compiling due to that error?